### PR TITLE
Fix allocator service tls auth for C# client and add a C# sample

### PR DESCRIPTION
**Pull request by pooneh-m**
**What type of PR is this?**
/kind bug

**What this PR does / Why we need it**:
C# client cannot talk to agone-allocator service due to language incompatibility for handling mTLS. Changed the mTLS handling to custom verify client certificate.

Also, added C# client sample code that is compatible with the golang gRPC server.

**Which issue(s) this PR fixes**:
Closes https://github.com/googleforgames/agones/issues/1421

### CODE REVIEW

File: examples/allocator-client/main.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func main() {
 	cacertFile := flag.String("cacert", "missing cacert", "the CA cert file for server signing certificate in PEM format")
 	externalIP := flag.String("ip", "missing external IP", "the external IP for allocator server")
 	namespace := flag.String("namespace", "default", "the game server kubernetes namespace")
-	multicluster := flag.Bool("multicluster", false, "enabling")
+	multicluster := flag.Bool("multicluster", false, "set to true for enabling the multi-cluster allocation")
```
Comment (markmandel) [inline examples/allocator-client/main.go:37]:
```suggestion
	multicluster := flag.Bool("multicluster", false, "set to true to enable multi-cluster allocation")
```

WDYT?

File: examples/allocator-client-csharp/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# A sample Allocator service C# client
```
Comment (markmandel) [inline examples/allocator-client-csharp/README.md:1]: Should we link this somewhere, such as here? 
https://agones.dev/site/docs/advanced/allocator-service/#send-allocation-request

(and wrapped in a feature code block)

File: examples/allocator-client/main.go (hunk 1 above)
Comment (pooneh-m) [reply examples/allocator-client/main.go]: Done. Thanks!

File: examples/allocator-client-csharp/README.md (hunk 2 above)
Comment (pooneh-m) [reply examples/allocator-client-csharp/README.md:1]: Thanks for pointing it out.
PTAL. I added a note.

File: site/content/en/docs/Advanced/allocator-service.md [hunk 3]
```diff
@@ -115,7 +115,8 @@ kubectl get pods -n agones-system -o=name | grep agones-allocator | xargs kubect
 
 ## Send allocation request
 
-Now the service is ready to accept requests from the client with the generated certificates. Create a [fleet](https://agones.dev/site/docs/getting-started/create-fleet/#1-create-a-fleet) and send a gRPC request to agones-allocator by providing the namespace to which the fleet is deployed. You can find the gRPC sample for sending allocation request at {{< ghlink href="examples/allocator-client/main.go" >}}allocator-client sample{{< /ghlink >}}.
+Now the service is ready to accept requests from the client with the generated certificates. Create a [fleet](https://agones.dev/site/docs/getting-started/create-fleet/#1-create-a-fleet) and send a gRPC request to agones-allocator. To start, take a look at the allocation gRPC client examples in {{< ghlink href="examples/allocator-client/main.go" >}}golang{{< /ghlink >}} and {{< ghlink href="examples/allocator-client-csharp/Program.cs" >}}C#{{< /ghlink >}} languages. In the following, the {{< ghlink href="examples/allocator-client/main.go" >}}golang gRPC client example{{< /ghlink >}} is used to allocate a Game Server in the default namespace.
```
Comment (markmandel) [inline site/content/en/docs/Advanced/allocator-service.md:118]: Do we want this to show up on the current documentation, or only after the next release?

If we want it after the next release, we should create publish and expire versions of this paragraph:
https://agones.dev/site/docs/contribute/#within-a-page

Other than that, this looks good to go 👍

Comment (pooneh-m) [reply site/content/en/docs/Advanced/allocator-service.md:123]: Thanks for the suggestion. Makes sense. I added the feature wrapper to expose the content with 1.6 release.

**Pull request merged**